docs/manual/nullness-checker.tex

```diff
@@ -456,6 +465,44 @@
 % TODO: explain more aspects, give more examples.
 
 
+\subsectionAndLabel{Nullness and conversions from collections to arrays}{nullness-collection-toarray}
+
+% Keep this in sync with the javadoc of org.checkerframework.checker.nullness.CollectionToArrayHeuristics
+
+The semantics of
+\sunjavadoc{java.base/java/util/Collection.html\#toArray(T[])}{Collection.toArray(T[])}
+cannot be captured by the nullness type system syntax. The nullness type of the
+returned array depends on the size of the passed parameter. In particular, the
+returned array component is of type \<@NonNull> if the following conditions
+hold:
+
+\begin{itemize}
+\item The receiver collection type argument is \<@NonNull>, and
+\item The passed array size is less than the collection size. Here are heuristics to
+       handle the most common cases:
+
+  \begin{itemize}
+    \item an empty array initializer, e.g. \<c.toArray(new String[] \{\})>,
```

Comment:
Why not combine the first two items to just be "the argument has length 0"?  Then elsewhere you can list what arrays are known to have size 0:  two specific patterns, plus a special case for those annotated as `@ArrayLen(0)`.